x-pack/plugins/viz_editor/public/pseudo_plugins/indexpattern_datasource_plugin/indexpattern_datasource_plugin.tsx

```diff
@@ -51,11 +52,46 @@ function DataPanel(props: PanelComponentProps<VisModel>) {
   );
 }
 
+function queryToEsAggsConfigs(query: Query): any {
+  return query.select.reverse().map((selectOperation, index) => {
```

Comment:
Reverse is a mutating method, so we're mutating our input here. This means that next time you run it, it will revert it to forward, and the next time to backward... Instead, we need to either use one of the functional utilities we discussed, or roll our own reverse function that is immutable.

Reply:
Good catch :)